node/api/routes.go

```diff
@@ -166,7 +167,7 @@ func cleanCloseHandler(next http.Handler) http.Handler {
 // UserAgent that contains the specified string.
 func RequireUserAgent(h http.Handler, ua string) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		if !strings.Contains(req.UserAgent(), ua) {
+		if !strings.Contains(req.UserAgent(), ua) && !whitelisted(req) {
```

Comment:
I think this is fine for now, though performance we should probably wrap each endpoint individually the same way that we do with `RequirePassword`, that way requests which don't require a user agent don't force us to do a strings.Contains every time another request is called.

With just one whitelisted endpoint, this shouldn't cause any performance issues, but if we start to adopt this as a pattern, it's a pattern that could cause problems if it gets out of hand.

Reply:
the advantage of a whitelist is that you get security by default, instead of needing to remember to add `RequirePassword` and `RequireUserAgent` manually every time you add a new endpoint.

Amazingly, I think this may be a place where `Context` comes in handy. It would allow us to whitelist routes by wrapping their handler in an `Unrestricted` function, like so:
```go
router.GET("/renter/stream/*siapath", Unrestricted(api.renterStreamHandler))

type unrestrictedContextKey struct {}

func Unrestricted(h http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
		req = req.WithContext(context.WithValue(req.Context(), unrestrictedContextKey{}, 0))
		h.ServeHTTP(w, req)
	}
}

func isUnrestricted(req *http.Request) bool {
	return req.Context().Value(unrestrictedContextKey{}) != nil
}

func RequireUserAgent(h http.Handler, ua string) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
		if !strings.Contains(req.UserAgent(), ua) && !isUnrestricted(req) {
			WriteError(w, Error{"Browser access disabled due to security vulnerability. Use Sia-UI or siac."}, http.StatusBadRequest)
			return
		}
		h.ServeHTTP(w, req)
	})
}
```